### PR TITLE
agent: update README.md for building from source

### PR DESCRIPTION
**Pull request by Apokleos**
agent: Update build instructions

Fix the instructions explaining how to build the agent from source now that `make` needs to be run to auto-generate some source files.

Fixes: #889.

Signed-off-by: LiYa'nan <oliverliyn@gmail.com>

### CODE REVIEW

File: src/agent/README.md [hunk 1]
```diff
@@ -43,12 +43,16 @@ The `rust-agent` depends on [`grpc-rs`](https://github.com/pingcap/grpc-rs) by P
...
 
 ### Build from Source
-The rust-agent need to be built with rust nightly, and static linked with musl.
+The rust-agent need to be built with static linked with musl:
 ```bash
-rustup target add x86_64-unknown-linux-musl
-git submodule update --init --recursive  
-sudo ln -s /usr/bin/g++ /bin/musl-g++  
-cargo build --target x86_64-unknown-linux-musl --release
+$ rustup target add x86_64-unknown-linux-musl
```
Comment (jodh-intel) [inline src/agent/README.md:48]: Kata works on multiple architectures, so how about:

```suggestion
$ arch=$(uname -m)
$ [ "$arch" == "ppc64le" ] && arch=powerpc64le
$ rustup target add "${arch}-unknown-linux-musl"
```
Note the ppc64 special-casing - see:

- https://github.com/kata-containers/tests/blob/2.0-dev/.ci/install_rust.sh
- https://github.com/kata-containers/tests/blob/master/.ci/kata-arch.sh

File: src/agent/README.md [hunk 2]
```diff
@@ -43,12 +43,16 @@ The `rust-agent` depends on [`grpc-rs`](https://github.com/pingcap/grpc-rs) by P
...
+$ sudo ln -s /usr/bin/g++ /bin/musl-g++  
+```
+Download the kata-containers source and build the agent:
+```bash
+$ go get -d -u github.com/kata-containers/kata-containers
```
Comment (jodh-intel) [inline src/agent/README.md:53]: This is slightly awkward. Although this is the simplest method to get the code, it assumes you have golang installed. But rust devs may not have. Also, we're trying to move away from using [`go get`](https://github.com/kata-containers/kata-containers/issues/823) for 2.0, so could you change this to:

```suggestion
$ GOPATH="${GOPATH:-$HOME/go}"
$ dir="$GOPATH/src/github.com/kata-containers"
$ mkdir -p "$dir" && cd "$dir"
$ git clone --depth 1 https://github.com/kata-containers/kata-containers && cd kata-containers
```

/cc @bergwolf

**Review by jodh-intel — COMMENTED** [review]
Thanks for raising @Apokleos !

I've added a couple of suggestions.

Also, please can you change your commit message slightly so that it conforms to https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format. Something like the following would work:

```
agent: Update build instructions

Fix the instructions explaining how to build the agent from source now that `make` needs to be run to auto-generate some source files.

Fixes: #889.

Signed-off-by: LiYa'nan <oliverliyn@gmail.com>
```

**Review by jodh-intel — COMMENTED** [review]
Thanks for raising @Apokleos !

I've added a couple of suggestions.

Also, please can you change your commit message slightly so that it conforms to https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format. Something like the following would work:

```
agent: Update build instructions

Fix the instructions explaining how to build the agent from source now that `make` needs to be run to auto-generate some source files.

Fixes: #889.

Signed-off-by: LiYa'nan <oliverliyn@gmail.com>
```

**Comment by Apokleos** [comment]
> Thanks for raising @Apokleos !
> 
> I've added a couple of suggestions.
> 
> Also, please can you change your commit message slightly so that it conforms to https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format. Something like the following would work:
> 
> ```
> agent: Update build instructions
> 
> Fix the instructions explaining how to build the agent from source now that `make` needs to be run to auto-generate some source files.
> 
> Fixes: #889.
> 
> Signed-off-by: LiYa'nan <oliverliyn@gmail.com>
> ```

Thank you @jodh-intel , I will change my message and PR

File: src/agent/README.md (hunk 1 above)
Comment (Apokleos) [reply src/agent/README.md:48]: Thank you for suggestions

**Comment by jodh-intel** [comment]
Thanks @Apokleos.

The branch is now conflicted so please can you rebase (and squash all commits into a single one)?

**Comment by Apokleos** [comment]
> Thanks @Apokleos.
> 
> The branch is now conflicted so please can you rebase (and squash all commits into a single one)?

Thank you @jodh-intel , I have done a new PR #895

File: src/agent/README.md [hunk 3]
```diff
@@ -43,12 +43,21 @@ The `rust-agent` depends on [`grpc-rs`](https://github.com/pingcap/grpc-rs) by P
...
-rustup target add x86_64-unknown-linux-musl
-git submodule update --init --recursive  
-sudo ln -s /usr/bin/g++ /bin/musl-g++  
-cargo build --target x86_64-unknown-linux-musl --release
+$ arch=$(uname -m)
+$ [ "$arch" == "ppc64le" ] && arch=powerpc64le
+$ rustup target add "${arch}-unknown-linux-musl"
+$ sudo ln -s /usr/bin/g++ /bin/musl-g++  
+```
+Download the kata-containers source and build the agent:
```
Comment (liubin) [inline src/agent/README.md:53]: It's better `s/kata-containers/Kata containers/`

**Comment by jodh-intel** [comment]
Closing this PR in favour of #895.

**Pull request closed without merging**